src/Adapter/Product/Options/RedirectTargetProvider.php

```diff
@@ -0,0 +1,151 @@
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\Product\Options;
+
+use PrestaShop\PrestaShop\Adapter\Category\Repository\CategoryPreviewRepository;
+use PrestaShop\PrestaShop\Adapter\LegacyContext;
+use PrestaShop\PrestaShop\Adapter\Product\Repository\ProductPreviewRepository;
+use PrestaShop\PrestaShop\Core\Domain\Category\Exception\CategoryNotFoundException;
+use PrestaShop\PrestaShop\Core\Domain\Category\ValueObject\CategoryId;
+use PrestaShop\PrestaShop\Core\Domain\Language\ValueObject\LanguageId;
+use PrestaShop\PrestaShop\Core\Domain\Product\Exception\ProductNotFoundException;
+use PrestaShop\PrestaShop\Core\Domain\Product\QueryResult\ProductRedirectTarget;
+use PrestaShop\PrestaShop\Core\Domain\Product\ValueObject\ProductId;
+use PrestaShop\PrestaShop\Core\Domain\Product\ValueObject\RedirectType;
+
+/**
+ * Build details on the product target based on the configuration (redirection type and entity id)
+ */
+class RedirectTargetProvider
+{
+    /**
+     * @var ProductPreviewRepository
+     */
+    private $productPreviewRepository;
+
+    /**
+     * @var CategoryPreviewRepository
+     */
+    private $categoryPreviewRepository;
+
+    /**
+     * @var LegacyContext
+     */
+    private $legacyContext;
+
+    /**
+     * @param ProductPreviewRepository $productPreviewRepository
+     * @param CategoryPreviewRepository $categoryPreviewRepository
+     * @param LegacyContext $legacyContext
+     */
+    public function __construct(
+        ProductPreviewRepository $productPreviewRepository,
+        CategoryPreviewRepository $categoryPreviewRepository,
+        LegacyContext $legacyContext
+    ) {
+        $this->productPreviewRepository = $productPreviewRepository;
+        $this->categoryPreviewRepository = $categoryPreviewRepository;
+        $this->legacyContext = $legacyContext;
+    }
+
+    /**
+     * @param string $redirectType
+     * @param int $redirectTargetId
+     *
+     * @return ProductRedirectTarget|null
+     *
+     * @throws CategoryNotFoundException
+     * @throws ProductNotFoundException
+     */
+    public function getRedirectTarget(
+        string $redirectType,
+        int $redirectTargetId
+    ): ?ProductRedirectTarget {
+        if (empty($redirectTargetId)) {
+            return null;
+        }
+
```

Comment:
```suggestion
```

Reply:
I did what you suggested at first without thinking too much about it But actually, it's important!!!
Maybe you did the same mistake as me and thought that it was equivalent to the default case that returns null

But actually, the point of this condition is that if no target ID is provided there is no point in looking for the target details, it's better for performance and it would probably not even work since we're looking for an entity with ID 0

Lucky for me my unit tests just reminded me that by kicking me in the face 😅

Reply:
Not blocking, but it could be nice to put this information into the code 😉 
```
if (empty($redirectTargetId)) {
    // if no target ID is provided then no need to look for the target details
    return null;
}
```